### ISSUE TITLE
the .so files in the armv8-apks are not compressed 

### ISSUE THREAD

**Issue by hanseul1795** [xamarin/xamarin-android#3629]
### Steps to Reproduce

1.  Generate the `.aab` file with Xamarin Android
2.  Create the armv7 and armv8 `.apks` with the bundletool (version 0.10.1)
3.  Compare the size of .so files within two `.apks` files.

### Expected Behavior
`.so` files in the` .apks` are compressed.

### Actual Behavior
`.so` files in the armv8 `.apks` are not compressed.

armv7 `.apks`  = ~25MB
armv8 `.apks` = ~75MB

### Version Information
Xamarin Android 10.0.0.40

I don't know if this issue is related to  Xamarin Android or the bundletool, I've posted this issue on both github just in case...

**Comment by brendanzagaeski** [comment]
Thanks for the report!

I believe you are seeing the correct expected behavior for bundletool. When deploying to a particular device, bundletool determines whether or not to compress the native libraries in the APK based on the app install location and the version of Android OS on the target device:

https://github.com/google/bundletool/blob/0.10.1/src/main/java/com/android/tools/build/bundletool/splitters/NativeLibrariesCompressionSplitter.java#L75-L86

In case it helps to think about the value of `shouldCompress` in the other direction, it means that when deploying to an Android 9.0 Pie (API level 28) or higher device, bundletool will always leave the native libraries uncompressed in the APK. When deploying to a device running an Android version between 6.0 Marshmallow (API level 23) and 8.1 Oreo (API level 27), bundletool will only leave the native libraries uncompressed in the APK if the app is not installable on external storage (for example, if the [`android:installLocation`](https://developer.android.com/guide/topics/manifest/manifest-element#install) `<manifest>` element attribute is set to `"internalOnly"`).

The reason to have uncompressed native libraries in the APK is that Android 6.0 Marshmallow and higher can read uncompressed native libraries directly from the APK instead of extracting them to the file system. See also the [`android:extractNativeLibs`](https://developer.android.com/guide/topics/manifest/application-element#extractNativeLibs) `<application>` element attribute.

If you are seeing a scenario where the compression of that native libraries does not follow this expected behavior, feel free to reply back on this issue to let us know. Thanks!

<details>
<summary>Additional details</summary>
I did a quick test with generating an APK Set archive, instead of generating an APK for a particular device.

```cmd
"C:\Program Files\Android\jdk\microsoft_dist_openjdk_1.8.0.25\bin\java.exe" -jar "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\Android\bundletool.jar" build-apks --aapt2="C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\Android\aapt2.exe" --bundle=com.contoso.androidapp1.aab --output=com.contoso.androidapp1.apks
```

In the APK Set archive generated by that command, the native libraries are *compressed* in both of the following nested archives:

  * *base-arm64_v8a.apk*
  * *base-armeabi_v7a.apk*

And the native libraries are *not* compressed in the following nested archives:

  * *base-arm64_v8a_2.apk*
  * *base-armeabi_v7a_2.apk*

In other words, when creating an APK Set archive, bundletool creates 2 versions of the base APK for both armeabi-v7a and arm64-v8a, where the difference between the 2 versions is whether the native libraries are compressed.
</details>

**Comment by trungnd90** [comment]
Hi, @brendanzagaeski do you know how to force compress native libraries in apk?

**Comment by trungnd90** [comment]
Hi @brendanzagaeski , do you know how to force compress native libraries in apk?

**Comment by brendanzagaeski** [comment]
In case any other users come across this issue when searching about this question, there is some additional information about the available options and trade-offs in some corresponding issues hanseul1795 filed:

* https://github.com/google/bundletool/issues/118#issuecomment-531474671
* https://github.com/google/bundletool/issues/119#issuecomment-532748348
